src/domain/ledger/index.types.d.ts

```diff
@@ -211,42 +227,56 @@ interface ILedgerService {
 
   getTransactionById(
     id: LedgerTransactionId,
-  ): Promise<LedgerTransaction | LedgerServiceError>
+  ): Promise<LedgerTransaction<WalletCurrency> | LedgerServiceError>
 
   getTransactionsByHash(
     paymentHash: PaymentHash | OnChainTxHash,
-  ): Promise<LedgerTransaction[] | LedgerServiceError>
+  ): Promise<LedgerTransaction<WalletCurrency>[] | LedgerServiceError>
 
   getTransactionsByWalletId(
     walletId: WalletId,
-  ): Promise<LedgerTransaction[] | LedgerServiceError>
+  ): Promise<LedgerTransaction<WalletCurrency>[] | LedgerServiceError>
```

Comment:
The `ByWalletId` query should / could probably be moved to `ByWalletDescriptor` so we can return the correct `LedgerTransaction<T>`? (not for this PR).

Reply:
Added to the PR notes in the description here for follow-up